Add docs for CREATE INDEX CONCURRENTLY (#12456)
yugabyte/yugabyte-db: docs/content/preview/api/ysql/the-sql-language/statements/ddl_create_index.md

| Parallelizes index loading? | yes | no |

To enable online schema migration for one `CREATE INDEX`, use `CREATE INDEX CONCURRENTLY`.

> 

Reviewer: Actually it feels a bit hard to understand which is the default behavior. I believe we are concurrent by default? Also I believe ysql_disable_index_backfill=true will take precedence over CREATE INDEX CONCURRENTLY? Would be good to mention that as well.

* Parallel writes are expected while creating the index, because concurrent builds for indexes on partitioned tables aren't supported. In this case, it's better to use concurrent builds to create indexes on each partition individually.
* [Row-level geo-partitioning](../../../../../explore/multi-region-deployments/row-level-geo-partitioning/) is being used. In this case, create the index separately on each partition to customize the tablespace in which each index is created.
* `CREATE INDEX CONCURRENTLY` is not supported for partitioned tables (see [CONCURRENTLY](#concurrently)).

### UNIQUE

Enforce that duplicate values in a table are not allowed.

### CONCURRENTLY
Enable online schema migration (see [Semantics](#semantics) for details), with some restrictions:

> * Upon creating an index on a temporary table, online schema migration will be disabled.

Reviewer 2: ```suggestion
* When creating an index on a temporary table, online schema migration is disabled.
```

> * `CREATE INDEX CONCURRENTLY` is not supported for partitioned table.

Reviewer 2: ```suggestion
* `CREATE INDEX CONCURRENTLY` is not supported for partitioned tables.
```
